pyuvdata/uvdata/miriad.py

```diff
@@ -41,1486 +41,1529 @@ def _pol_to_ind(self, pol):
+            default is True, meaning the optional parameters will be checked).
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters before
+            writing the file (the default is True, meaning the acceptable
+            range check will be done).
+        strict_uvw_antpos_check : bool
+            Option to raise an error rather than a warning if the check that
+            uvws match antenna positions does not pass.
+        no_antnums : bool
+            Option to not write the antnums variable to the file.
+            Should only be used for testing purposes.
+
+        Raises
+        ------
+        ValueError
+            If the frequencies are not evenly spaced or are separated by more
+            than their channel width.
+            The `phase_type` of the object is "unknown".
+        TypeError
+            If any entry in extra_keywords is not a single string or number.
 
         """
-        # check if telescope name is present
-        if self.telescope_name is None:
-            self._load_miriad_variables(uv)
+        from . import aipy_extracts
+
+        # change time_array and lst_array to mark beginning of integration,
+        # per Miriad format
+        miriad_time_array = self.time_array - self.integration_time / (24 * 3600.0) / 2
+        if self.telescope_location is not None:
+            latitude, longitude, altitude = self.telescope_location_lat_lon_alt_degrees
+            miriad_lsts = uvutils.get_lst_for_time(
+                miriad_time_array, latitude, longitude, altitude
+            )
+
+        # Miriad requires j>i which we call ant1<ant2
+        self.conjugate_bls(convention="ant1<ant2")
+
+        if run_check:
+            self.check(
+                check_extra=check_extra,
+                run_check_acceptability=run_check_acceptability,
+                check_freq_spacing=True,
+                strict_uvw_antpos_check=strict_uvw_antpos_check,
+            )
+
+        if os.path.exists(filepath):
+            if clobber:
+                print("File exists: clobbering")
+                shutil.rmtree(filepath)
+            else:
+                raise IOError("File exists: skipping")
+
+        uv = aipy_extracts.UV(filepath, status="new")
+
+        # initialize header variables
+        uv._wrhd("obstype", "mixed-auto-cross")
+        # avoid inserting extra \n.
+        if not self.history[-1] == "\n":
+            self.history += "\n"
+        uv._wrhd("history", self.history)
+
+        # recognized miriad variables
+        uv.add_var("nchan", "i")
+        uv["nchan"] = self.Nfreqs
+        uv.add_var("npol", "i")
+        uv["npol"] = self.Npols
+
+        #####################################################
+        # Frequency information here
+        uv.add_var("nspect", "i")
+        uv["nspect"] = self.Nspws
+
+        if self.future_array_shapes:
+            freq_array_use = self.freq_array
```

Comment:
You might need to similarly add something like this for `channel_width` -- it may only get used when `flex_spw=True`, but the array shape will of course be different when `future_array_shapes=True`.

Reply:
Whoops -- I take that back. `channel_width` is already of shape `("Nfreqs",)`... my bad!